Simplify cards json

AmunMRa opened the pull request:
Removed conditionals and flattened properties to apply to all cards. Descriptions clarify nuance. Added image ratio and image proportions.

<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

Cards json was not being ingested properly into design table on v0 website due to unsupported conditional properties. Simplifies the schema to meet design intent. Relies more on descriptions for nuance between card variants.

## How Has This Been Tested?

I validated the schema changes locally using the Moon workspace pipeline. After updating cards.json, I ran:

`pnpm moon check`


This installed the pinned pnpm version (10.17.1), resolved workspace dependencies, and executed the configured validation tasks. The pipeline reported all checks passing, confirming that the updated schema is valid and doesn’t introduce breaking issues.

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

AmunMRa committed:
Simplify cards json

Removed conditionals and flattened properties to apply to all cards. Descriptions clarify nuance. Added image ratio and image proportions.

GarthDB commented:
<!-- moon-run-report: unknown -->

## Run report for [608b635d](https://github.com/adobe/spectrum-tokens/commit/608b635d1e3a5dd35cc95123cfb410f7d5ca9194)
Total time: 27s | Comparison time: 28s | Estimated savings: 992.3ms (3.5% faster)
|     | Action | Time | Status | Info |
| :-: | :----- | ---: | :----- | :--- |
| ⬛️ | `SetupProto(0.51.4)` | 0.1ms | Skipped |  |
| 🟩 | `SyncWorkspace` | 7ms | Passed |  |
| 🟩 | `SyncProject(site)` | 0.3ms | Passed |  |
| 🟩 | `SyncProject(componentSchemas)` | 0.4ms | Passed |  |
| 🟩 | `RunTask(site:copyComponentSchema)` | 637.6ms | Passed |  |
| 🟩 | `RunTask(site:copyTokenTypeSchema)` | 644.8ms | Passed |  |
| ⬛️ | `SetupToolchain(node:~20.12)` | 1.4s | Skipped |  |
| 🟩 | `InstallWorkspaceDeps(node:~20.12)` | 7.5s | Passed |  |
| 🟩 | `RunTask(site:clean)` | 179.9ms | Passed |  |
| 🟥 | `RunTask(componentSchemas:test)` | 2.9s | Failed |  |
| 🟩 | `RunTask(site:build)` | 18s | Passed |  |
| 🟩 | `RunTask(site:export)` | 105.6ms | Passed |  |

<details><summary><strong>Touched files</strong></summary><div>

```
packages/component-schemas/schemas/components/cards.json
```

</div></details>

AmunMRa committed:
Update cards.json

AmunMRa committed:
Added styling and fixed required props

Added prop for styling and required needed to be updated to correct prop.

AmunMRa committed:
Added title prop and description prop

Add prop to allow for optional title and description within cards

GarthDB committed:
fix(component-schemas): add changeset for cards schema fix

- Add remark-frontmatter plugin to preserve YAML frontmatter
- Update lint-staged config to use frontmatter plugin first
- Fix changeset format to use patch version for schema fix

GarthDB committed:
fix(component-schemas): add changeset for cards schema fix

- Mark as patch version since this fixes a previous issue
- Exclude changeset files from remark processing to preserve frontmatter
- This resolves the CI/CD pipeline failure for PR #620
